### PR TITLE
migrate `prefect_aws.batch` off `sync_compatible`

### PR DESCRIPTION
**Pull request by zzstoatzz**
related to #15008

migrates `prefect_aws.batch` tasks to explicit sync / async via `async_dispatch`

also updates `async_dispatch` to allow dispatching async task objects

will continue with `prefect_aws` after this PR

### CODE REVIEW

File: src/integrations/prefect-aws/tests/test_batch.py [hunk 1]
```diff
@@ -74,8 +84,45 @@ def test_flow():
...
+    job_queue_arn, job_definition_arn, aws_credentials
+):
+    job_id = batch_submit(
+        "batch_test_job",
+        job_queue_arn,
+        job_definition_arn,
+        aws_credentials,
+        _sync=True,
+    )
+    assert_valid_job_id(job_id)
```
Comment (zzstoatzz) [inline src/integrations/prefect-aws/tests/test_batch.py:128]: i tried a couple permutations of pytest tricks but explicitly writing out the individual cases felt simplest to me

File: src/prefect/_internal/compatibility/async_dispatch.py [hunk 2]
```diff
@@ -19,6 +22,14 @@ def is_in_async_context() -> bool:
         return False
 
 
+def _is_acceptable_callable(obj: Union[Callable, Task]) -> bool:
+    if inspect.iscoroutinefunction(obj):
+        return True
+    if isinstance(obj, Task) and inspect.iscoroutinefunction(obj.fn):
+        return True
+    return False
```
Comment (desertaxle) [inline src/prefect/_internal/compatibility/async_dispatch.py:30]: Looks like we'll need to bump the version of `prefect` in `prefect-aws` to get this change?

File: src/integrations/prefect-aws/prefect_aws/batch.py [hunk 3]
```diff
@@ -3,22 +3,22 @@
...
 from prefect.logging import get_run_logger
-from prefect.utilities.asyncutils import run_sync_in_worker_thread, sync_compatible
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
 from prefect_aws.credentials import AwsCredentials
 
 
 @task
-@sync_compatible
-async def batch_submit(
+async def batch_submit_async(
```
Comment (desertaxle) [inline src/integrations/prefect-aws/prefect_aws/batch.py:13]: Should this be `abatch_submit`?

File: src/integrations/prefect-aws/tests/test_batch.py (hunk 1 above)
Comment (desertaxle) [reply src/integrations/prefect-aws/tests/test_batch.py:128]: The smallest of nits, but can you group the tests into classes so it's easier to find the tests for each implementation?

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**